x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/ReservedRolesStore.java

```diff
@@ -768,7 +768,7 @@ public static RoleDescriptor kibanaSystemRoleDescriptor(String name) {
                         "logs-*",
                         "synthetics-*",
                         "traces-*",
-                        "/metrics-.*&~(metrics-endpoint\\.metadata_current_default)/",
+                        "/metrics-.*&~(metrics-endpoint\\.metadata_current_default.*)/",
```

Comment:
I think this should be `/metrics-.*&~(metrics-endpoint\\.metadata_current_default*` in order to be backwards compatible with legacy indices in the Endpoint package.

This also has me thinking that another privilege would be needed here.  I believe we need `PutMappingAction` for the index aliases that are created.  One index that isn't accounted for is, for instance `.metrics-endpoint.metadata_united_default.all`.  I think we should also add `.metrics-endpoint.metadata_united_default*` to account for that.  Let me know your thoughts.

Reply:
This line is different to the others because it's a regex, so `.*` in this line is effectively like `*` in the others.

> I believe we need `PutMappingAction` for the index aliases that are created.

I think that's `IndicesAliasesAction` which is granted on line 827.

@qn895 do we ever put mappings directly on the transform destination indices? Or do we always use an index template? If we ever directly set mappings directly then more changes are needed in this file.

> I think we should also add `.metrics-endpoint.metadata_united_default*` to account for that.

I don't think it's needed because it begins with a dot. This line is saying the permissions on line 778 should be applied to `metrics-*` indices _unless_ they also match `metrics-endpoint.metadata_current_default*`. So `.metrics-endpoint.metadata_current_default*` and `.metrics-endpoint.metadata_united_default*` don't match because they start with a dot.

Then all of `metrics-endpoint.metadata_current_default*`, `.metrics-endpoint.metadata_current_default*` and `.metrics-endpoint.metadata_united_default*` instead get the permissions on line 827.

Reply:
> > I believe we need `PutMappingAction` for the index aliases that are created.

> I think that's `IndicesAliasesAction` which is granted on line 827.

> @qn895 do we ever put mappings directly on the transform destination indices? Or do we always use an index template? If we ever directly set mappings directly then more changes are needed in this file.

Got it thanks for the clarification.  I defer to @qn895 for more info.

> I don't think it's needed because it begins with a dot. This line is saying the permissions on line 778 should be applied to `metrics-*` indices _unless_ they also match `metrics-endpoint.metadata_current_default*`. So `.metrics-endpoint.metadata_current_default*` and `.metrics-endpoint.metadata_united_default*` don't match because they start with a dot.

> Then all of `metrics-endpoint.metadata_current_default*`, `.metrics-endpoint.metadata_current_default*` and `.metrics-endpoint.metadata_united_default*` instead get the permissions on line 827.

Got it makes sense, thanks for the clarification

Reply:
> @qn895 do we ever put mappings directly on the transform destination indices? Or do we always use an index template? If we ever directly set mappings directly then more changes are needed in this file.

We do not put mappings directly on the transform destination indices, just the index & component templates. The only thing we set directly when creating these indices is the alias.

Reply:
> We do not put mappings directly on the transform destination indices, just the index & component templates.

OK great. I think the permissions we've got now should work then.